src/test/scala/reporter/CloudWatchReporterSpec.scala

```diff
@@ -0,0 +1,35 @@
+package reporter
+
+import com.blacklocus.metrics.CloudWatchReporterBuilder
+import com.codahale.metrics.{Metric, MetricFilter, MetricRegistry}
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{FlatSpec, Matchers, PrivateMethodTester}
+import Utils.buildMetricFilter
+import scala.util.matching.Regex
+
+
+class CloudWatchReporterSpec extends FlatSpec with Matchers with PrivateMethodTester with MockFactory {
+
+  private val metricRegistry: MetricRegistry = new MetricRegistry
+  private val metric: Metric = mock[Metric]
+
+
+  "Metrics filter" should "match any metric when empty string filter is given" in {
+    val filter = buildMetricFilter("")
+
+    filter.matches("any_metrics_name", metric) should be(true)
+    filter.matches("xfaewojz", metric) should be(true)
+  }
+
+  "Metrics filter" should "match metrics that have the regex pattern: true" in {
+    val filter = buildMetricFilter("([a-zA-Z]+.[a-zA-Z].*LFS+-loader-.+.lag)")
+
+    filter.matches("gauge.readings_V1-LFS-loader-aws.lag", metric) should be(true)
+  }
+
+   "Metrics filter" should "match metrics that have the regex pattern: false" in {
+    val filter = buildMetricFilter("([a-zA-Z]+.[a-zA-Z].*LFS+-loader-.+.lag)")
+
+    filter.matches("gauge.vers.stop.lag", metric) should be(false)
+  }
```

Comment:
Could you add a test for when an invalid regexp is passed as an input string?

Reply:
created a test of what happens if theres an invalid regexp since it would fail there instead of when its inserted to the method